### PR TITLE
Update Routing Logic and Improve SignUp Process

### PR DESCRIPTION
**Pull request by JPraskala**
KAN-196

This pull request implements several key changes to enhance the routing and user experience in the application. The updates include modifications to the routing structure and updates to the sign-up functionality.

#### Changes Made:
- **App.js Enhancements:**
  - Updated the signup route to redirect users to the home page (`/`) after a successful sign-up instead of the dashboard.
  - Simplified the route declaration by consolidating the route component for the signup page into a single line for better readability.

- **Functions.js Updates:**
  - Removed the automatic page reload (`window.location.reload()`) after a successful sign-up response. This change is aimed at preventing loss of state and promoting smoother transitions once a user signs up.

- **SignUpForm.js Adjustments:**
  - Integrated the `useNavigate` hook from `react-router-dom` to programmatically navigate the user to the homepage (`/`) instead of relying on a page reload.
  - Added a call to `navigate('/')` immediately after user sign-up to provide immediate feedback and guide users to the appropriate next steps in the application.

These changes aim to streamline the user experience after signing up and improve the overall flow of the application, ensuring users are correctly directed without unnecessary page reloads.

### CODE REVIEW

File: src/blocks/SignUpForm.js [hunk 1]
```diff
@@ -36,6 +38,7 @@ const SignUpForm = ({ toggleForm }) => {
     }
     //signUp function imported from Functions.js
     signUp(firstName, lastName, password, email, role);
+    navigate('/');
```
Comment (amhajja) [inline src/blocks/SignUpForm.js:41]: `signUp` is doing an asynchronous HTTP request, so you don't want to navigate until we make sure the HTTP request is completed. 

We can make `signUp` return a promise: 
```
return axios.post...
```
That way, we can do `signUp(firstName, lastName, password, email, role).then((response) => ...`

Alternatively, you can move the axios call to `apiService.js` and remove the code in `Functions.js` altogether, which probably is the more consistent and better approach.

**Review by amhajja — CHANGES_REQUESTED** [review]

File: src/blocks/SignUpForm.js (hunk 1 above)
Comment (AustinfHunter) [reply src/blocks/SignUpForm.js]: Jacob moved the  axios call to `apiSerivce.js` before I took this issue off his hands.

File: src/blocks/SignUpForm.js [hunk 2]
```diff
@@ -102,41 +112,47 @@ const SignUpForm = ({ toggleForm }) => {
...
-                label="Password *"
+                name="password"
+                label="Password"
                 type="password"
                 variant="outlined"
-                value={password}
-                onChange={(e) => setPassword(e.target.value)}
+                value={formData.password}
+                onChange={(e) => handleFormChange(e)}
+                required
```
Comment (amhajja) [inline src/blocks/SignUpForm.js:135]: I like those changes!

**Review by amhajja — APPROVED** [review]
This is nice!

**Pull request merged**